.github/workflows/smoke-tests.yml

```diff
@@ -0,0 +1,105 @@
+name: "System Smoke Tests"
+on:
+  push:
+    # run on pushes to master or release/*
+    branches:    
+      - master
+      - release/*
+  pull_request:
+    # run on pull requests targeting master
+    branches:
+      - master
+jobs:
+  setup:
+    runs-on: self-hosted
+    steps:
+    - name: Check Test Cluster
+      run: |
+        if kubectl get nodes; then
+          kubectl get nodes
+        else
+          echo "test cluster not active. Attempting to create through kind..."
+          kind delete cluster 2>/dev/null
+          kind create cluster
+          kubectl get nodes
+        fi
+  test-source:
+    runs-on: self-hosted
+    needs: setup
+    steps:
+    - uses: actions/checkout@v2
+    - name: Clean Cluster
+      run: |
+        kubectl delete all --all --wait=true --force --grace-period=0
+        sleep 5
```

Comment:
This is not good. `all` indeed will not clean everything.
If the costs aren't too high, I recommend deleting a cluster and creating again.
This almost certainly is going to leak something.
It's also a security problem to reuse a cluster between patches.

Reply:
creating a new cluster isn't too difficult if it's better practice. Done